torch/csrc/jit/python/pybind_utils.h

```diff
@@ -291,9 +292,9 @@ inline InferredType tryToInferType(py::handle input) {
   py::bool_ isEnumValue = py::isinstance(input, enum_type);
   if (py::cast<bool>(isEnumValue)) {
     auto enum_class = input.attr("__class__");
-    auto enum_type =
-        py::cast<TypePtr>(py::module::import("torch.jit.annotations")
-                              .attr("try_ann_to_type")(enum_class, py::none()));
+    auto enum_type = py::cast<TypePtr>(
+        py::module::import("torch.jit.annotations")
+            .attr("try_ann_to_type")(enum_class, SourceRange()));
```

Comment:
Can we get an accurate `SourceRange` for this? It is used later to construct a call stack for the error message in case recursive class compilation fails.

Reply:
This is actually something I wanted to seek suggestion from you. 

This code path is ultimately reached from:
https://github.com/pytorch/pytorch/blob/dab9bbfce77f10d1b120c495709de251067e76d7/torch/jit/_recursive.py#L103

You can see that somehow fake_range() is used multiple times in that function, which I do not fully understand why. Does it imply somehow there is no valid source range at this callsite?

Reply:
On lines 301-316, TorchScript classes are handled slightly differently. Instead of using `try_ann_to_type`, the code looks up the JIT type of the class in the compilation unit directly and returns that. Could we do something similar?

My educated guess is that we cannot, because I noticed that all of your enum test classes are not explicitly scripted with `@torch.jit.script` (is that for better usability?). Consequently, we need to script them during type inference.

Reply:
Please correct me if this is wrong. The reason I am not following the logic in "class" branch is that we may not have necessarily compiled (and added enum class to jit._state). Thus we can't get type of enum without having "try_ann_to_type" actually verifies/compiles the enum class object.

@eellison could you confirm?

As for "why not decorating enum classes with `@torch.jit.script`, it is exactly like what you said, I don't think user need to decorate it to ensure eager/script interoperability as long as re-create same-identity object. And there is no reference semantic issue because enums are always constant. Let me know what you think, especially with regard to consistency with API of class redesign.

Reply:
I don't think the SourceRange is used as part of the type at all, so it should be fine use an empty / fake source range in this function call